tests/e2e_tests/utils.py

```diff
@@ -123,18 +125,37 @@ def call_add_proposal(substrate: SubstrateInterface, wallet: bittensor.wallet) -
     return response.is_success
 
 
-def wait_interval(interval, subtensor):
+async def wait_epoch(subtensor, netuid=1):
+    q_tempo = [
+        v.value
+        for [k, v] in subtensor.query_map_subtensor("Tempo")
+        if k.value == netuid
+    ]
+    if len(q_tempo) == 0:
+        raise Exception("could not determine tempo")
+    tempo = q_tempo[0]
+    logging.info(f"tempo = {tempo}")
+    await wait_interval(tempo, subtensor, netuid)
+
+
+async def wait_interval(tempo, subtensor, netuid=1):
+    interval = tempo + 1
     current_block = subtensor.get_current_block()
-    next_tempo_block_start = (current_block - (current_block % interval)) + interval
+    last_epoch = current_block - 1 - (current_block + netuid + 1) % interval
```

Comment:
Hey man. What's the reason for the netuid here? @ibraheem-opentensor and I were wondering.

Reply:
It's the algorithm as used in subtensor!

Reply:
I'm not making this up ;-)

Reply:
It only accidentally worked before, and only for the first few epochs before the differences start to matter

Reply:
Cool. Assumed you had a good reason for it. Just didn't seem immediately apparent. Thanks.